src/App.tsx

```diff
@@ -16,36 +18,55 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [filter, setFilter] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleFilter = (action: string) => {
+    switch (action) {
+      case 'alphabet':
+        setFilter('alphabet');
+        break;
+      case 'length':
+        setFilter('length');
+        break;
+      case 'reverse':
+        setIsReversed(prev => !prev);
+        break;
+      case 'reset':
+        setFilter('');
+        setGoods(goodsFromServer);
+        setIsReversed(false);
+        break;
+      default:
+        break;
+    }
+  };
+
+  useEffect(() => {
+    let sortedGoods = [...goodsFromServer];
+
+    if (filter === 'alphabet') {
+      sortedGoods = sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (filter === 'length') {
+      sortedGoods = sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      sortedGoods.reverse();
+    }
+
+    setGoods(sortedGoods);
+  }, [filter, isReversed]);
```

Comment:
try to rewrite it without using the state for goods

we can just 
```
const goods = getSortedGoods(goodsFromServer, filter, isReversed)
```
and move all logic to that function